src/charging-station/Bootstrap.ts

```diff
@@ -154,13 +156,37 @@ export default class Bootstrap {
         poolOptions: {
           workerChoiceStrategy: Configuration.getWorkerPoolStrategy(),
         },
-        messageHandler: async (msg: ChargingStationWorkerMessage) => {
-          if (msg.id === ChargingStationWorkerMessageEvents.STARTED) {
-            this.uiServer.chargingStations.add(msg.data.id as string);
-          } else if (msg.id === ChargingStationWorkerMessageEvents.STOPPED) {
-            this.uiServer.chargingStations.delete(msg.data.id as string);
-          } else if (msg.id === ChargingStationWorkerMessageEvents.PERFORMANCE_STATISTICS) {
-            await this.storage.storePerformanceStatistics(msg.data as unknown as Statistics);
+        messageHandler: (msg: InternalChargingStationWorkerMessage) => {
+          const workerEventStarted = (data: ChargingStationData) => {
+            this.uiServer.chargingStations.set(data.hashId, data.data);
+            ++this.numberOfChargingStations;
+          };
+          const workerEventStopped = (data: ChargingStationData) => {
+            this.uiServer.chargingStations.delete(data.hashId);
+            --this.numberOfChargingStations;
```

Comment:
Stopping a charging station from the main process will need more work: the message structure is broadcasted to worker and does not contains any information about the CS to stop.